crates/bevy_asset/src/io/gated.rs

```diff
@@ -65,7 +62,7 @@ impl<R: AssetReader> AssetReader for GatedReader<R> {
         let receiver = {
             let mut gates = self.gates.write();
             let gates = gates
-                .entry(path.to_path_buf())
+                .entry(path.into())
```

Comment:
```suggestion
                .entry_ref(path.as_ref())
```
You might be able to elide the the allocation entirely by using `entry_ref`.